Sources/App/routes.swift

```diff
@@ -57,7 +57,7 @@ func routes(_ app: Application) throws {
     do {  // package pages
         do {  // temporary, hacky docc-proxy
             app.get(":owner", ":repository", ":reference", "documentation") {
-                try await PackageController.documentation(req: $0, fragment: .documentationRedirect)
+                try await PackageController.documentation(req: $0)
```

Comment:
In fact, the better way to deal with this is to simply route to a different handler that is much simpler that what we have in our current `documentation(req:fragment:)` handler - it doesn't even need to call out to AWS at all - and that's also why the bad url doesn't even have any effect I believe. We were making an unnecessary AWS call when hitting the root documentation URL without an `:archive` path, ignoring entirely what it did, if I'm reading this right.

Reply:
This also trims down the long `documentation(req:fragment:)` function a bit.

Reply:
You're correct, we don't need to hit AWS for this. Only look whether there are `docArchives` and redirect.